tests/unit-tests/test_nn_model.py

```diff
@@ -516,6 +519,614 @@ def test_rgcn_with_edge_features(input_dim, output_dim, dev):
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r1', 'n1')).fc_src.weight)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r1', 'n1')).fc_dst.weight)
+    layer = layer.to(dev)
+
+    emb8 = layer(block, node_feats, edge_feats)
+    assert_almost_equal(emb8_woefeat['n1'].detach().cpu().numpy(),
+                        emb8['n1'].detach().cpu(), decimal=3)
+
+    # 'sub' operator, same outputs
+    layer = RelationalAttLayer(
+        input_dim, output_dim, etypes,
+        num_heads=2,
+        edge_feat_mp_op='sub',
+        edge_feat_name={("n0", "r0", "n1"): ['feat'], ("n0", "r1", "n1"): ['feat']},
+        bias=False, activation=None, self_loop=False, dropout=0.0, norm=None)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r0', 'n1')).fc_src.weight)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r0', 'n1')).fc_dst.weight)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r1', 'n1')).fc_src.weight)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r1', 'n1')).fc_dst.weight)
+    layer = layer.to(dev)
+
+    emb8 = layer(block, node_feats, edge_feats)
+    assert_almost_equal(emb8_woefeat['n1'].detach().cpu().numpy(),
+                        emb8['n1'].detach().cpu(), decimal=3)
+
+    # 'mul' operator, different outputs
+    layer = RelationalAttLayer(
+        input_dim, output_dim, etypes,
+        num_heads=2,
+        edge_feat_mp_op='mul',
+        edge_feat_name={("n0", "r0", "n1"): ['feat'], ("n0", "r1", "n1"): ['feat']},
+        bias=False, activation=None, self_loop=False, dropout=0.0, norm=None)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r0', 'n1')).fc_src.weight)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r0', 'n1')).fc_dst.weight)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r1', 'n1')).fc_src.weight)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r1', 'n1')).fc_dst.weight)
+    layer = layer.to(dev)
+
+    emb8 = layer(block, node_feats, edge_feats)
+    assert np.not_equal(emb8_woefeat['n1'].detach().cpu().numpy(),
+                        emb8['n1'].detach().cpu().numpy()).any()
+
+    # 'div' operator, different outputs
+    layer = RelationalAttLayer(
+        input_dim, output_dim, etypes,
+        num_heads=2,
+        edge_feat_mp_op='div',
+        edge_feat_name={("n0", "r0", "n1"): ['feat'], ("n0", "r1", "n1"): ['feat']},
+        bias=False, activation=None, self_loop=False, dropout=0.0, norm=None)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r0', 'n1')).fc_src.weight)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r0', 'n1')).fc_dst.weight)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r1', 'n1')).fc_src.weight)
+    th.nn.init.ones_(layer.conv._get_module(('n0', 'r1', 'n1')).fc_dst.weight)
+    layer = layer.to(dev)
+
+    emb8 = layer(block, node_feats, edge_feats)
+    assert np.not_equal(emb8_woefeat['n1'].detach().cpu().numpy(),
+                        emb8['n1'].detach().cpu().numpy()).any()
+
+
```

Comment:
As discussed in our weekly meeting, please remove the `main` function and all subfunctions before merge.